temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java

```diff
@@ -135,7 +137,19 @@ public Builder setPollThreadNamePrefix(String pollThreadNamePrefix) {
 
     public PollerOptions build() {
       if (uncaughtExceptionHandler == null) {
-        uncaughtExceptionHandler = (t, e) -> log.error("uncaught exception", e);
+        uncaughtExceptionHandler =
+            (t, e) -> {
+              if (e instanceof RuntimeException
+                  && e.getCause() != null
+                  && e.getCause() instanceof StatusRuntimeException) {
+                StatusRuntimeException sre = (StatusRuntimeException) e.getCause();
+                if (sre.getStatus().getCode() == Status.Code.INVALID_ARGUMENT) {
+                  log.info("Uncaught exception", e);
```

Comment:
As we know the exact meaning of the exception I would change the log message to something more descriptive.